pre_commit/commands/run.py

```diff
@@ -72,13 +72,19 @@ def get_filenames(args, include_expr, exclude_expr):
 
 
 def _run_single_hook(hook, repo, args, write, skips=frozenset()):
-    filenames = get_filenames(args, hook['files'], hook['exclude'])
+    filenames = []
+    # if the hook is marked as always_run, do not compute the files to run
+    # in that case, simply run the script once not matter the changes
+    compute_file_names = 'always_run' not in hook or not hook['always_run']
```

Comment:
I almost want this to be less special and still compute the filenames.  That'd be more in line with the rest of the project and I think make things simpler to reason about.